kimchi/src/circuits/polynomials/keccak.rs

```diff
@@ -21,65 +77,265 @@ use crate::circuits::{
-    /// calls the Keccak rotation gadget
-    pub fn create_keccak(new_row: usize) -> (usize, Vec<Self>) {
-        // Initial Generic gate to constrain the prefix of the output to be zero
-        let mut gates = vec![CircuitGate::<F>::create_generic_gadget(
-            Wire::for_row(new_row),
-            GenericGateSpec::Pub,
-            None,
-        )];
-        Self::create_keccak_rot(&mut gates, new_row + 1, new_row)
-    }
+pub const RC: [u64; 24] = [
+    0x0000000000000001,
+    0x0000000000008082,
+    0x800000000000808a,
+    0x8000000080008000,
+    0x000000000000808b,
+    0x0000000080000001,
+    0x8000000080008081,
+    0x8000000000008009,
+    0x000000000000008a,
+    0x0000000000000088,
+    0x0000000080008009,
+    0x000000008000000a,
+    0x000000008000808b,
+    0x800000000000008b,
+    0x8000000000008089,
+    0x8000000000008003,
+    0x8000000000008002,
+    0x8000000000000080,
+    0x000000000000800a,
+    0x800000008000000a,
+    0x8000000080008081,
+    0x8000000000008080,
+    0x0000000080000001,
+    0x8000000080008008,
+];
+
+//~
+//~ | `KeccakRound` | [0...440) | [440...1540) | [1540...2344) |
+//~ | ------------- | --------- | ------------ | ------------- |
+//~ | Curr          | theta     | pirho        | chi           |
+//~
+//~ | `KeccakRound` | [0...100) |
+//~ | ------------- | --------- |
+//~ | Next          | iota      |
+//~
+//~ -----------------------------------------------------------------------------------------------------------------------------------------------------------------------
+//~
+//~ | Columns  | [0...100) | [100...120) | [120...200) | [200...220) | [220...240) | [240...260)  | [260...280) | [280...300)  | [300...320)  | [320...340) | [340...440) |
+//~ | -------- | --------- | ----------- | ----------- | ----------- | ----------- | ------------ | ----------- | ------------ | ------------ | ----------- | ----------- |
+//~ | theta    | state_a   | state_c     | shifts_c    | dense_c     | quotient_c  | remainder_c  | bound_c     | dense_rot_c  | expand_rot_c | state_d     | state_e     |
+//~
+//~ | Columns  | [440...840) | [840...940) | [940...1040) | [1040...1140) | [1140...1240) | [1240...1340) | [1340...1440) | [1440...1540) |
+//~ | -------- | ----------- | ----------- | ------------ | ------------- | ------------- | ------------- | ------------- | ------------- |
+//~ | pirho    | shifts_e    | dense_e     | quotient_e   | remainder_e   | bound_e       | dense_rot_e   | expand_rot_e  | state_b       |
+//~
+//~ | Columns  | [1540...1940) | [1940...2340) | [2340...2344 |
+//~ | -------- | ------------- | ------------- | ------------ |
+//~ | chi      | shifts_b      | shifts_sum    | f00          |
+//~
+//~ | Columns  | [0...4) | [4...100) |
+//~ | -------- | ------- | --------- |
+//~ | iota     | g00     | rest_g    |
+//~
+#[derive(Default)]
+pub struct KeccakRound<F>(PhantomData<F>);
```

Comment:
It is a single round of the permutation? (i.e. it is going to be called 24 times?)